src/main.js

```diff
@@ -1624,6 +1624,41 @@ function discoveryFeedCardsFromQueries(proxies) {
     }).catch(e => logError(e, 'Query failed'));
 }
 
+
+// A helper class for tracking single-connections to signals
+//
+// A bit gross in the sense that it uses O(N) lookups, but
+// JS doesn't seem to have unique IDs for objects.
+const SignalManager = new Lang.Class({
```

Comment:
I think this class is more confusing than anything to be honest... It's a pretty well-established practice to store signal connection IDs in the main class like that, and this adds cognitive overhead for little gain IMO. I'd rather keep it simple.

Reply:
Yeah ... hmmm. I appreciate that the practice exists but I really don't like it ;-) I kind of see it as boilerplate.

I wonder if there's a better way to do this rather than explicitly checking every single signal to be `== -1` every time we want to reconnect it?

Reply:
The practice is only really needed when the object emitting signals outlives the signal receiver, which is not the case most of the time. In C it's possible to use `g_signal_connect_object()` to make things simpler; not sure if that works from JS.
Either way, this is also a different kind of boilerplate ;-) I'm not arguing against having a better convenience for that use case in JS if possible, but I wouldn't want to special-case it here.